lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -654,37 +654,44 @@ pcmk__instance_is_compatible(const pe_resource_t *instance,
     return true;
 }
 
+/*!
+ * \internal
+ * \brief Find an instance that matches a given resource by node and role
+ *
+ * \param[in] match_rsc  Resource that instance must match (for logging only)
+ * \param[in] rsc        Clone or bundle resource to check for matching instance
+ * \param[in] node       Instance must match this node
+ * \param[in] role       If not RSC_ROLE_UNKNOWN, instance must match this role
+ * \param[in] current    If true, compare instance's original node and role,
+ *                       otherwise compare assigned next node and role
+ *
+ * \return \p rsc instance matching \p node and \p role if any, otherwise NULL
+ */
 static pe_resource_t *
-find_compatible_child_by_node(const pe_resource_t *local_child,
-                              const pe_node_t *local_node,
-                              const pe_resource_t *rsc, enum rsc_role_e filter,
-                              gboolean current)
+find_compatible_instance_on_node(const pe_resource_t *match_rsc,
+                                 const pe_resource_t *rsc,
+                                 const pe_node_t *node, enum rsc_role_e role,
+                                 bool current)
 {
-    GList *gIter = NULL;
-    GList *children = NULL;
+    GList *instances = NULL;
 
-    if (local_node == NULL) {
-        crm_err("Can't colocate unrunnable child %s with %s", local_child->id, rsc->id);
-        return NULL;
-    }
-
-    crm_trace("Looking for compatible child from %s for %s on %s",
-              local_child->id, rsc->id, pe__node_name(local_node));
-
-    children = get_instance_list(rsc);
-    for (gIter = children; gIter != NULL; gIter = gIter->next) {
-        pe_resource_t *child_rsc = (pe_resource_t *) gIter->data;
+    instances = get_instance_list(rsc);
+    for (GList *iter = instances; iter != NULL; iter = iter->next) {
+        pe_resource_t *instance = (pe_resource_t *) iter->data;
 
-        if (pcmk__instance_is_compatible(child_rsc, local_node, filter,
-                                         current)) {
-            crm_trace("Pairing %s with %s on %s",
-                      local_child->id, child_rsc->id, pe__node_name(local_node));
-            return child_rsc;
+        if (pcmk__instance_is_compatible(instance, node, role, current)) {
+            pe_rsc_trace(match_rsc,
+                         "Found %s instance %s compatible with %s on %s",
+                         rsc->id, instance->id, match_rsc->id,
+                         pe__node_name(node));
```

Comment:
optionally, could add the role to the log message if not `RSC_ROLE_UNKNOWN`; ditto below for none found

Reply:
good idea